assignments/array-methods.js

```diff
@@ -3,81 +3,98 @@
+  {"id":48,"first_name":"Anderea","last_name":"MacGiolla Pheadair","email":"amacgiollapheadair1b@xing.com","shirt_size":"2XL","company_name":"Kwimbee","donation":214},
+  {"id":49,"first_name":"Bel","last_name":"Alway","email":"balway1c@ow.ly","shirt_size":"S","company_name":"Voolia","donation":107},
+  {"id":50,"first_name":"Shell","last_name":"Baine","email":"sbaine1d@intel.com","shirt_size":"M","company_name":"Gabtype","donation":171}];
 
 // ==== Challenge 1: Use .forEach() ====
-// The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
+// The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName.
 let fullName = [];
+
+runners.forEach (runner => fullName.push(`${runner.first_name} ${runner.last_name}`) );
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
-console.log(allCaps); 
+let allCaps = runners.map(runner => runner.first_name.toUpperCase() );
+console.log(allCaps);
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+let largeShirts = runners.filter( runner => runner.shirt_size == 'L' );
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
+let ticketPriceTotal = runners.reduce( (donation, runner) => donation += runner.donation, 0 );
 console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to solve 3 unique problems using one or many of the array methods listed above.
 
 // Problem 1
+//Last Name, First Name sorted alphabetically for a roll sheet
+let sortedRunners = runners.map( runner => `${runner.last_name}, ${runner.first_name}`).sort();
+console.log(sortedRunners);
 
 // Problem 2
+//list of participating businesses
+let theBusinesses = [];
+runners.forEach( runner => {
+  if(!theBusinesses.includes(runner.company_name)) {
+    theBusinesses.push(runner.company_name);
+  }
+});
+theBusinesses.sort();
+console.log(theBusinesses);
 
-// Problem 3
\ No newline at end of file
+// Problem 3
```

Comment:
Awesome work in this file! Great creativity and commenting for your work.

For your filter, why not just do?:
`runner => runner.company_name !== 'Photospace')`

Reply:
I'm going to be completely honest. It's been so long since I've had to or need to use !== that I completely forgot that it existed.